dockers/docker-sflow/port_index_mapper.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+""" port_index_mapper
+    A mapper service that watches for NetLink NEWLINK and DELLINKs
+    to construct a PORT_INDEX_TABLE in state DB which includes the
+    interface name, the interface index and the ifindex.
+
+    Note : Currently supports only interfaces supported by port_util.
+"""
+
+import os
+import sys
+import syslog
+import signal
+import traceback
+from pyroute2 import IPDB
+from swsssdk import SonicV2Connector, port_util
+
+RTM_NEWLINK = 16
+RTM_DELLINK = 17
+
+PORT_INDEX_TABLE_NAME = 'PORT_INDEX_TABLE'
+SYSLOG_IDENTIFIER = 'port_index_mapper'
+
+ipdb = None
+state_db = None
+
+def interface_callback(ipdb, nlmsg, action):
+    global state_db
+
+    try:
+        msgtype = nlmsg['header']['type']
+        if (msgtype != RTM_NEWLINK and msgtype != RTM_DELLINK):
+            return
+
+        # filter out unwanted messages
+        change = nlmsg['change']
+        if (change != 0xFFFFFFFF):
+            return
+    
+        attrs = nlmsg['attrs']
+        for list in attrs:
+            if list[0] == 'IFLA_IFNAME':
+                ifname = list[1]
+                break
+            else:
+                return
+    
+        # Extract the port index from the interface name
+        index = port_util.get_index_from_str(ifname)
+        if index is None:
+            return
+    
+        _hash = '{}|{}'.format(PORT_INDEX_TABLE_NAME, ifname)
+    
+        if msgtype == RTM_NEWLINK:
+            state_db.set(state_db.STATE_DB, _hash, 'index', str(index))
```

Comment:
I think its better to put updating STATE DB to a separate function, as it is called from multiple places.